File: 0xxx/118-PascalsTriangle/118.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        if (numRows == 1)
            return {{1}};
        vector<vector<int>> ans = generate(numRows - 1);
        vector<int> row = {1};
        for (int i = 1; i < ans.back().size(); i++)
            row.push_back(ans.back()[i - 1] + ans.back()[i]);
        row.push_back(1);
        ans.push_back(row);
        return ans;
    }
};

File: 0xxx/118-PascalsTriangle/118.py
a = [[1]]
for numRows in range(29):
    new = [1] + [a[-1][i - 1] + a[-1][i] for i in range(1, len(a[-1]))] + [1]
    a.append(new)
    print(f"else if (numRows == {numRows + 2}) return {a};".replace("[", "{").replace("]", "}"))
